# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/webrender_bindings/RenderThread.cpp
# Commit: 4d0b3e4d35b9
# Full Hash: 4d0b3e4d35b99b99f65c8e7308d314054ab33a8d
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-12-20 04:10:48
# Regressor Bug: 1829026
# File Overlap Count: 1
# Description:
#   Bug 1829026 - Don't force sync for async-present if GL not thread-safe. r=aosmond
#   
#   If GL is not thread-safe, then the check for whether we use the canvas render thread
#   is not accurate. The thread-safe check takes precedence, and if we are not thread-safe,
#   we are actually on the render thread. It should be safe to not force sync here since
# ==============================================================================

diff -r b1c8877fa4c8 -r 4d0b3e4d35b9 gfx/webrender_bindings/RenderThread.cpp
--- a/gfx/webrender_bindings/RenderThread.cpp	Tue Dec 19 20:29:45 2023 +0000
+++ b/gfx/webrender_bindings/RenderThread.cpp	Tue Dec 19 20:44:27 2023 +0000
@@ -1356,7 +1356,12 @@
     mProgramsForCompositorOGL->Clear();
     mProgramsForCompositorOGL = nullptr;
   }
-  mShaders = nullptr;
+  if (mShaders) {
+    if (mSingletonGL) {
+      mSingletonGL->MakeCurrent();
+    }
+    mShaders = nullptr;
+  }
   mSingletonGL = nullptr;
 }
 
